app/modules/ads/bill-the-lizard.js

```diff
@@ -0,0 +1,91 @@
+import targeting from './targeting';
+
+const cheshirecatPredictions = {};
+
+let cheshirecatConfig = null;
+let cheshirecatCalled = false;
+let biddersCalled = false;
+
+export default {
+  configureBillTheLizard(instantGlobals) {
+    const { context, slotService } = window.Wikia.adEngine;
+    const { billTheLizard } = window.Wikia.adServices;
+
+    if (context.get('bidders.prebid.bidsRefreshing.enabled')) {
+      cheshirecatConfig = instantGlobals.wgAdDriverBillTheLizardConfig || {};
+
+      context.set('services.billTheLizard.projects', cheshirecatConfig.projects);
+      context.set('services.billTheLizard.timeout', cheshirecatConfig.timeout || 0);
+
+      billTheLizard.projectsHandler.enable('cheshirecat');
+      billTheLizard.executor.register('catlapseIncontentBoxad', () => {
+        slotService.disable(this.getNextIncontentId(cheshirecatPredictions), 'catlapsed');
+      });
+
+      context.set('bidders.prebid.bidsRefreshing.bidsBackHandler', this.callCheshireCat.bind(this));
+      context.push('listeners.slot', {
+        onRenderEnded: (adSlot) => {
+          if (adSlot.config.slotName === 'incontent_boxad_1' && biddersCalled && !cheshirecatCalled) {
+            this.callCheshireCat();
+          }
+        },
+      });
+    }
+  },
+
+  callCheshireCat() {
+    const { context } = window.Wikia.adEngine;
+    const { billTheLizard } = window.Wikia.adServices;
+
+    context.set('services.billTheLizard.parameters.cheshirecat', this.serializeBids('mobile_in_content'));
+
+    cheshirecatCalled = true;
+
+    billTheLizard.call(['cheshirecat'])
+      .then((predictions) => {
+        const identifier = this.getNextIncontentId(cheshirecatPredictions);
+        const prediction = Object.keys(predictions).map(key => `${key}=${predictions[key]}`).join(';');
+
+        cheshirecatPredictions[identifier] = prediction;
+        context.set(`services.billTheLizard.parameters.cheshirecatSlotResponses.${identifier}`, prediction);
+      });
+  },
+
+  getNextIncontentId(predictions) {
```

Comment:
I'd define it outside export, it's not needed in billTheLizard interface

Reply:
Done.